pkg/apis/networking/validation/validation.go

```diff
@@ -408,3 +464,15 @@ func validateIngressClassParameters(params *api.TypedLocalObjectReference, fldPa
 
 	return allErrs
 }
+
+// allPathsPassRegexValidation returns true if the Ingress has paths that all
+// match the Ingress path validation with requireRegexPath enabled.
+func allPathsPassRegexValidation(ingress *networking.Ingress) bool {
+	for _, rule := range ingress.Spec.Rules {
+		errs := validateIngressRuleValue(&rule.IngressRuleValue, field.NewPath(""), IngressValidationOptions{requireRegexPath: true})
+		if len(errs) > 0 {
+			return false
+		}
```

Comment:
I'd recommend isolating this to just the path regex check... validateIngressRuleValue does other validation

```suggestion
		if rule.HTTP == nil {
			continue
		}
		for _, path := range rule.HTTP.Paths {
			if len(path) == 0 {
				continue
			}
			if _, err := regexp.CompilePOSIX(rule.Path); err != nil {
				return false
			}
		}
```

Reply:
Updated, thanks!